kong/plugins/hmac-auth/access.lua

```diff
@@ -27,8 +29,71 @@ local DIGEST = "digest"
 local SIGNATURE_NOT_VALID = "HMAC signature cannot be verified"
 local SIGNATURE_NOT_SAME = "HMAC signature does not match"
 
+local new_tab
+do
+  local ok
+  ok, new_tab = pcall(require, "table.new")
+  if not ok then
+    new_tab = function() return {} end
+  end
+end
+
 local _M = {}
 
+local hmac = {
+  ["hmac-sha1"] = function(secret, data)
+    return crypto.hmac.digest("sha1", data, secret, true)
+  end,
+  ["hmac-sha256"] = function(secret, data)
+    return crypto.hmac.digest("sha256", data, secret, true)
+  end,
+  ["hmac-sha384"] = function(secret, data)
+    return crypto.hmac.digest("sha384", data, secret, true)
+  end,
+  ["hmac-sha512"] = function(secret, data)
+    return crypto.hmac.digest("sha512", data, secret, true)
+  end
+}
+
+local function list_as_set(list)
+  local set = new_tab(#list, 0)
```

Comment:
Are you sure @thibaultcha, because in following code, 1st argument is length?

https://github.com/Mashape/kong/blob/master/kong/plugins/ip-restriction/handler.lua#L26